ngraph/python/src/ngraph/utils/node_factory.py

```diff
@@ -37,9 +39,23 @@ def create(
 
         @return   Node object representing requested operator with attributes set.
         """
+        if arguments is None and attributes is None:
+            node = self.factory.create(op_type_name)
+            node._attr_cache = {}
+            node._attr_cache_valid = False
+            return node
+
+        if arguments is None and attributes is not None:
+            raise UserInputError(
+                'Cannot create "{}" op without arguments. Provide arguments along with attributes.'.format(
+                    op_type_name)
+            )
```

Comment:
I am not sure if this is entirely true. As we can create empty op and set some `attributes` without setting `arguments`. Two possibilities would be:

1) Notify user that there is a need to validate node afterwards in case of passing only attributes via some logger warning.
2) Keep it this way and throw error.

@ilyachur what do you think? Which scenario aligns better to c++?

Reply:
I agree, assuming that we can create an empty op without providing arguments user shall be informed that validate function shall be called before op object is used. @ilyachur what is your opinion?

Reply:
I am not familiar with operation creation in python API, but in C++ we can create an operation without arguments and after that use `set_arguments()` method to specialize operation inputs.
Am I right that this code prohibits such behavior?

Reply:
In general this function is taking 2 optional parameters: `arguments` and `attributes`. Depends on theirs value we have following scenarios:
1. `arguments = None` and `attributes = None` -> empty object op is created
2. `arguments` and `attributes = None` -> create op without attributes
3. `arguments` and `attributes` -> create op with attributes
**4. `arguments = None` and `attributes` -> return error as no arguments were provided.**

What shall be the behavior for the 4th option?
- allow to build op without arguments and notify user that validate() shall be triggered
- return error (as in current implementation)
- create an empty op and notify user to add arguments and attributes and after that run validate()
- any other proposal...

Reply:
I think in any case where user has partially defined op he should call evaluate after full definition.

In C++ we have next cases:

- arguments and attributes are not defined (default constructor)
- Create operation with arguments and attributes.

Maybe we can support only these cases in Python.

Reply:
Applied.